core/sr-io/without_std.rs

```diff
@@ -938,7 +943,20 @@ impl CryptoApi for () {
 	fn secp256k1_ecdsa_recover(sig: &[u8; 65], msg: &[u8; 32]) -> Result<[u8; 64], EcdsaVerifyError> {
 		let mut pubkey = [0u8; 64];
 		match unsafe {
-			ext_secp256k1_ecdsa_recover.get()(msg.as_ptr(), sig.as_ptr(), pubkey.as_mut_ptr())
+			ext_secp256k1_ecdsa_recover.get()(0, msg.as_ptr(), sig.as_ptr(), pubkey.as_mut_ptr())
+		} {
+			0 => Ok(pubkey),
+			1 => Err(EcdsaVerifyError::BadRS),
+			2 => Err(EcdsaVerifyError::BadV),
+			3 => Err(EcdsaVerifyError::BadSignature),
+			_ => unreachable!("`ext_secp256k1_ecdsa_recover` only returns 0, 1, 2 or 3; qed"),
```

Comment:
You return `4` in the implementation as well.

Reply:
yeah but it can never be reached, hence the `unreachable`

Reply:
outdated.